numalogic/tools/data.py

```diff
@@ -9,19 +9,39 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import logging
+from typing import Optional, Generator, Iterator
 
 import numpy as np
+import numpy.typing as npt
+import pytorch_lightning as pl
 import torch
-from numalogic.tools.exceptions import DataModuleError, InvalidDataShapeError
-from numpy.typing import NDArray
 from pytorch_lightning.utilities.types import TRAIN_DATALOADERS, EVAL_DATALOADERS
 from torch import Tensor
 from torch.utils.data import IterableDataset, DataLoader
-import pytorch_lightning as pl
+
+from numalogic.tools.exceptions import InvalidDataShapeError
+
+_LOGGER = logging.getLogger(__name__)
 
 
 class StreamingDataset(IterableDataset):
-    def __init__(self, data: NDArray[float], seq_len: int):
+    r"""
+    An iterable Dataset designed for streaming time seris input.
+
+    Args:
+        data: A numpy array containing the input data in the shape of (batch, num_features).
+        seq_len: Length of the sliding window sequences to be generated from the input data
+
+    Raises:
+        ValueError: If the sequence length is greater than the data size
+        InvalidDataShapeError: If the input data array does not
```

Comment:
Where is this InvalidDataShapError raised?

Reply:
Line number 49

Reply:
Got it. Thought I had expanded all the lines but missed that part.